docs/dgram.md

```diff
@@ -0,0 +1,98 @@
+ZJS API for UDP datagram sockets
+================================
+
+* [Introduction](#introduction)
+* [Web IDL](#web-idl)
+* [API Documentation](#api-documentation)
+* [Client Requirements](#requirements)
+* [Sample Apps](#sample-apps)
+
+Introduction
+------------
+The `dgram` API is based on the subset of the
+[corresponding module](https://nodejs.org/api/dgram.html) in Node.js.
+It allows to send and receive UDP datagrams.
+
+Web IDL
+-------
+This IDL provides an overview of the interface; see below for documentation of
+specific API functions.
+
+```javascript
+// require returns a socket factory object
+// var dgram = require('dgram');
+
+[NoInterfaceObject]
+interface dgram {
+    DgramSocket createSocket(string udp4_or_udp6);
+};
+
+[NoInterfaceObject]
+interface DgramSocket {
+    void on(string event, RecvCallback cb);
+    void bind(int port, string ip_addr);
+    void send(Buffer buf, unsigned long offset, unsigned long len, int port, string ip_addr, [SendCallback cb]);
+    void close();
+};
+
+callback RecvCallback = void (Buffer msg, RemoteInfo rinfo);
+callback SendCallback = void (Error err);  // or undefined if no error
+
+
+callback EventCallback = void (various);  // callback arg depends on event
+
+dictionary RemoteInfo {
+    string ip_addr;
+    string family;
+    int port;
+};
+```
+
+API Documentation
+-----------------
+### dgram.createSocket
+
+`DgramSocket createSocket(string type);`
+
+Create a datagram socket of given type, which must be `'udp4'` or `'udp6'`.
+
+### DgramSocket.on
+
+`void on(string event, RecvCallback callback);`
+
```

Comment:
Would it be possible to support the error event also since I anticipate it will be very useful given the state of the Zephyr net stack and this new API.